Encounter Me/Pages/TrailProgress.razor

```diff
@@ -183,15 +188,32 @@
     [JSInvokable]
     public async void TrailFinished()
     {
-        Guid userId = Guid.Parse(authState.User.FindFirst(ClaimTypes.UserData).Value);
-        currentUser = await userDataService.GetUserDetails(userId);
-
         //enables success modal
         ModalControl.OpenModal(_successModal);
         //used to update UI elements
         StateHasChanged();
         //int xpGain = LevelAndXp.XpFromTrail(_trailInProgress);
         int xpGain = _capturePointInProgress.DefenseLevel * 50 + 100;
         userDataService.UpdateUserXp(currentUser, xpGain);
+
+        if (_capturePointInProgress.faction == 0)
+        {
+            _capturePointInProgress.faction = (int)(currentUser.Faction) + 1;
+            _capturePointInProgress.DefenseLevel = 1;
+        }
+        else if (_capturePointInProgress.faction == ((int)(currentUser.Faction) + 1))
+        {
+            if (_capturePointInProgress.DefenseLevel < 5)
+                _capturePointInProgress.DefenseLevel++;
```

Comment:
I have the same little remark about hardcoded numbers - imagine you decide to change the maximum defence level in the future from 5 to 6. What do you do then? If you remember where this little number was - great, but what if you don't? My idea for the future - I think it would be best if the control of `capturePoint` state (whether it is captured, what is the defence level, what is maximum defence level, lessening/increasing of the defence level after trail is finished) would be relayed to separate class.